Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm

```diff
@@ -15062,42 +15121,89 @@ - (void)dealloc
     [super dealloc];
 }
 
-- (WKFoundTextPosition *)start
+- (BOOL)isEmpty
+{
+    return NO;
+}
+
+- (WebKit::WebFoundTextRange)webFoundTextRange
 {
-    WKFoundTextPosition *position = [WKFoundTextPosition textPositionWithOffset:self.location order:self.order];
-    return position;
+    return { };
 }
 
-- (UITextPosition *)end
+@end
+
+@implementation WKFoundTextPosition
+@end
+
+@implementation WKFoundDOMTextRange
+
+- (WKFoundDOMTextPosition *)start
 {
-    WKFoundTextPosition *position = [WKFoundTextPosition textPositionWithOffset:(self.location + self.length) order:self.order];
+    WKFoundDOMTextPosition *position = [WKFoundDOMTextPosition textPositionWithOffset:self.location order:self.order];
     return position;
 }
 
-- (BOOL)isEmpty
+- (WKFoundDOMTextPosition *)end
 {
-    return NO;
+    WKFoundDOMTextPosition *position = [WKFoundDOMTextPosition textPositionWithOffset:(self.location + self.length) order:self.order];
+    return position;
 }
 
 - (WebKit::WebFoundTextRange)webFoundTextRange
 {
-    return { self.location, self.length, self.frameIdentifier, self.order };
+    WebKit::WebFoundTextRange::DOMData data { self.location, self.length };
+    return { data, self.frameIdentifier, self.order };
 }
 
 @end
 
-@implementation WKFoundTextPosition
+@implementation WKFoundDOMTextPosition
 
-+ (WKFoundTextPosition *)textPositionWithOffset:(NSUInteger)offset order:(NSUInteger)order
++ (WKFoundDOMTextPosition *)textPositionWithOffset:(NSUInteger)offset order:(NSUInteger)order
 {
-    auto pos = adoptNS([[WKFoundTextPosition alloc] init]);
+    auto pos = adoptNS([[WKFoundDOMTextPosition alloc] init]);
```

Comment:
Nit: I know this is old code, but `RetainPtr pos = ...`?

Reply:
Done.